lib/libc/stdlib/malloc/mrs/mrs.c

```diff
@@ -280,30 +300,48 @@ static struct mrs_quarantine offload_quarantine; /* quarantine for the offload t
 
 static void *mrs_calloc_bootstrap(size_t number, size_t size);
 
-static inline __attribute__((always_inline)) void mrs_puts(const char *p)
+static inline __attribute__((always_inline)) void
+mrs_puts(const char *p)
 {
 	size_t n = strlen(p);
 	write(2, p, n);
 }
 
 /* locks */
 
-#define mrs_lock(mtx) do {if (pthread_mutex_lock((mtx))) {mrs_puts("pthread error\n");exit(7);}} while (0)
-#define mrs_unlock(mtx) do {if (pthread_mutex_unlock((mtx))) {mrs_puts("pthread error\n");exit(7);}} while (0)
+#define mrs_lock(mtx) do {						\
+	if (pthread_mutex_lock((mtx)) != 0) {				\
+		mrs_puts("pthread error\n");				\
+		exit(7);						\
+	}								\
+} while (0)
+
+#define mrs_unlock(mtx) do {						\
+	if (pthread_mutex_unlock((mtx)) != 0) {				\
+		mrs_puts("pthread error\n");				\
+		exit(7);						\
+	}								\
+} while (0)
 
 /*
  * hack to initialize mutexes without calling malloc. without this, locking
  * operations in allocation functions would cause an infinite loop. the buf
  * size should be at least sizeof(struct pthread_mutex) from thr_private.h
  */
-#define create_lock(name) \
-	pthread_mutex_t name; \
-	char name ## _buf[256] __attribute__((aligned(16))); \
-	void *name ## _storage() { \
-		return name ## _buf; \
+#define create_lock(name)						\
+	pthread_mutex_t name;						\
+	char name ## _buf[256] __attribute__((aligned(16)));		\
+									\
+	void *								\
+	name ## _storage()						\
```

Comment:
```suggestion
	name ## _storage(void)						\
```
I think newer compilers will whine about this otherwise?

Reply:
Oof, yes.  I fixed several of those in other places but missed this one.

Reply:
Oh, it's actually terrible.  The function takes two size_t arguments and fixing this actually breaks the build.  Have to add the actual args.